src/account_multisig.ts

```diff
@@ -159,54 +159,96 @@ export class Account2FA extends AccountMultisig {
+            const newArgs = Buffer.from(JSON.stringify({ 'num_confirmations': 2 }));
             actions.push(functionCall('new', newArgs, MULTISIG_GAS, MULTISIG_DEPOSIT),)
         }
         console.log('deploying multisig contract for', accountId)
-        return await super.signAndSendTransactionWithAccount(accountId, actions);
+        const result = await super.signAndSendTransactionWithAccount(accountId, actions);
+        // check if newLocalPublicKey was added to account, if not whole tx failed
+        const newLocalPublicKeyStr = newLocalPublicKey.toString()
+        accessKeys = await this.getAccessKeys()
+        if (!accessKeys.find(({ public_key }) => public_key === newLocalPublicKeyStr)) {
+            throw new Error('Error deploying multisig contract')
+        }
+        return result
     }
 
-    async disable(contractBytes: Uint8Array) {
+    async disable(contractBytes: Uint8Array, newLocalPublicKey: PublicKey) {
         const { accountId } = this
-        const accessKeys = await this.getAccessKeys()
-        const lak2fak = accessKeys.filter(({ access_key }) => 
+        let [
+            confirmOnlyKey,
+            localPublicKey,
+            accessKeys
+        ] = await Promise.all([
+            this.postSignedJson('/2fa/getAccessKey', { accountId }),
+            this.connection.signer.getPublicKey(accountId, this.connection.networkId),
+            this.getAccessKeys()
+        ])
+        confirmOnlyKey = PublicKey.from(confirmOnlyKey.publicKey)
+        const localPublicKeyStr = localPublicKey.toString()
+        const lak2fak = accessKeys.filter(({ public_key, access_key }) => 
+            public_key !== localPublicKeyStr &&
             access_key && access_key.permission && access_key.permission.FunctionCall &&
             access_key.permission.FunctionCall.receiver_id === accountId &&
             access_key.permission.FunctionCall.method_names &&
             access_key.permission.FunctionCall.method_names.length === 4 &&
             access_key.permission.FunctionCall.method_names.includes('add_request_and_confirm')    
         )
-        const confirmOnlyKey = PublicKey.from((await this.postSignedJson('/2fa/getAccessKey', { accountId })).publicKey)
+        // newLocalPublicKey and lak2fak -> full access keys, existing localStorageKey is deleted
         const actions = [
             deleteKey(confirmOnlyKey),
+            deleteKey(localPublicKey),
+            addKey(newLocalPublicKey, null),
             ...lak2fak.map(({ public_key }) => deleteKey(public_key)),
             ...lak2fak.map(({ public_key }) => addKey(public_key, null)),
             deployContract(contractBytes),
         ]
         console.log('disabling 2fa for', accountId)
-        return await this.signAndSendTransaction(accountId, actions)
+        const result = await this.signAndSendTransaction(accountId, actions)
+        // check if newLocalPublicKey was added to account, if not whole tx failed
```

Comment:
@mattlockyer not sure I understand this logic? `signAndSendTransaction` should be throwing error when transaction failed. Is there some case where it doesn't throw properly? Basically how to trigger this code?